pkg/apis/keycloak/v1alpha1/keycloakbackup_types.go

```diff
@@ -7,16 +7,53 @@ import (
 // KeycloakBackupSpec defines the desired state of KeycloakBackup
 // +k8s:openapi-gen=true
 type KeycloakBackupSpec struct {
-	Restore bool            `json:"restore,omitempty"`
-	AWS     KeycloakAWSSpec `json:"aws,omitempty"`
+	// Controls automatic restore behavior.
+	// Currently not implemented (yet)
+	// +optional
+	Restore bool `json:"restore,omitempty"`
+	// If provided, an automatic database backup will be created on AWS S3.
+	// +optional
+	AWS KeycloakAWSSpec `json:"aws,omitempty"`
 }
 
 // KeycloakAWSSpec defines the desired state of KeycloakBackupSpec
 // +k8s:openapi-gen=true
 type KeycloakAWSSpec struct {
+	// Provides a secret name used for encrypting database data.
+	// The secret needs to be in the following form:
+	//
+	//     apiVersion: v1
+	//     kind: Secret
+	//     metadata:
+	//       name: <Secret name>
+	//     type: Opaque
+	//     stringData:
+	//       GPG_PUBLIC_KEY: <GPG Public Key>
+	//       GPG_TRUST_MODEL: <GPG Trust Model>
+	//       GPG_RECIPIENT: <GPG Recipient>
+	//
+	// For more information, please refer to the manual pages.
```

Comment:
documentation

Reply:
Add that if specified database backup is encrypted, if not specified it's not

Reply:
Fixed.